From 9ece3f0d7c0fa3327605fb452a181340fcf602da Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Tue, 11 Aug 2026 09:14:40 +0000
Subject: [PATCH] Revert "Remove deprecated IRadio <1.4 APIs and references"

This reverts commit 4d53aa0caec7029bf6c9b5789c1708bc4bb71b76.

[Port to A16] TelephonyManager/ITelephony anchors (shutdownRadios javadoc) intact. RILConstants: SETUP_DATA_AUTH_PAP_CHAP anchor removed upstream; LCE constants placed next to LCE_NOT_SUPPORTED instead.

Change-Id: I024b02110622cc600c9dfd36542e22318a4f8438
---
 .../android/telephony/TelephonyManager.java   | 23 +++++++++++++++++++
 .../internal/telephony/ITelephony.aidl        | 11 +++++++++
 .../internal/telephony/RILConstants.java      |  5 ++++
 3 files changed, 39 insertions(+)

diff --git a/telephony/java/android/telephony/TelephonyManager.java b/telephony/java/android/telephony/TelephonyManager.java
index a433703856f6..0a2d318377e0 100644
--- a/telephony/java/android/telephony/TelephonyManager.java
+++ b/telephony/java/android/telephony/TelephonyManager.java
@@ -11687,6 +11687,29 @@ public class TelephonyManager {
         return result;
     }
 
+    /**
+     * Returns the result and response from RIL for oem request
+     *
+     * @param oemReq the data is sent to ril.
+     * @param oemResp the respose data from RIL.
+     * @return negative value request was not handled or get error
+     *         0 request was handled succesfully, but no response data
+     *         positive value success, data length of response
+     * @hide
+     * @deprecated OEM needs a vendor-extension hal and their apps should use that instead
+     */
+    @Deprecated
+    public int invokeOemRilRequestRaw(byte[] oemReq, byte[] oemResp) {
+        try {
+            ITelephony telephony = getITelephony();
+            if (telephony != null)
+                return telephony.invokeOemRilRequestRaw(oemReq, oemResp);
+        } catch (RemoteException ex) {
+        } catch (NullPointerException ex) {
+        }
+        return -1;
+    }
+
     /**
      * Shut down all the live radios over all the slot indexes.
      *
diff --git a/telephony/java/com/android/internal/telephony/ITelephony.aidl b/telephony/java/com/android/internal/telephony/ITelephony.aidl
index ca7574b71aa4..0b67c7b02422 100644
--- a/telephony/java/com/android/internal/telephony/ITelephony.aidl
+++ b/telephony/java/com/android/internal/telephony/ITelephony.aidl
@@ -1064,6 +1064,17 @@ interface ITelephony {
             in List<String> gsmNonRoamingList, in List<String> cdmaRoamingList,
             in List<String> cdmaNonRoamingList);
 
+    /**
+     * Returns the result and response from RIL for oem request
+     *
+     * @param oemReq the data is sent to ril.
+     * @param oemResp the respose data from RIL.
+     * @return negative value request was not handled or get error
+     *         0 request was handled succesfully, but no response data
+     *         positive value success, data length of response
+     */
+    int invokeOemRilRequestRaw(in byte[] oemReq, out byte[] oemResp);
+
     /**
      * Check if any mobile Radios need to be shutdown.
      *
diff --git a/telephony/java/com/android/internal/telephony/RILConstants.java b/telephony/java/com/android/internal/telephony/RILConstants.java
index 20717b0c306f..f3ed462a643f 100644
--- a/telephony/java/com/android/internal/telephony/RILConstants.java
+++ b/telephony/java/com/android/internal/telephony/RILConstants.java
@@ -74,6 +74,11 @@ public interface RILConstants {
     int SIM_SAP_CONNECT_OK_CALL_ONGOING = 35;
     int LCE_NOT_SUPPORTED = 36;                     /* Link Capacity Estimation (LCE) not
                                                        supported */
+
+    /* LCE service related constants. */
+    int LCE_NOT_AVAILABLE = -1;
+    int LCE_STOPPED = 0;
+    int LCE_ACTIVE = 1;
     int NO_MEMORY = 37;                             /* Not sufficient memory to process the
                                                        request */
     int INTERNAL_ERR = 38;                          /* Hit unexpected vendor internal error
-- 
2.43.0

